error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/Slf4jLogDeclarationTest.java

```diff
@@ -0,0 +1,178 @@
+            "    // BUG: Diagnostic contains:",
+            "    private static Logger LOG = LoggerFactory.getLogger(VariableMissingFinal.class);",
+            "  }",
+            "",
+            "  static class VariableMissingPrivate {",
+            "    // BUG: Diagnostic contains:",
+            "    static final Logger LOG = LoggerFactory.getLogger(VariableMissingPrivate.class);",
+            "  }",
+            "",
+            "  static class VariableMissingStatic {",
+            "    // BUG: Diagnostic contains:",
+            "    private final Logger LOG = LoggerFactory.getLogger(VariableMissingStatic.class);",
+            "  }",
+            "",
+            "  static class WrongVariableName {",
+            "    // BUG: Diagnostic contains:",
+            "    private static final Logger BAR = LoggerFactory.getLogger(WrongVariableName.class);",
+            "  }",
+            "",
+            "  static class WrongArgumentGetLogger {",
+            "    // BUG: Diagnostic contains:",
+            "    private static final Logger LOG = LoggerFactory.getLogger(String.class);",
+            "  }",
+            "",
+            "  interface InterfaceWithNoCanonicalModifiers {",
+            "    Logger LOG = LoggerFactory.getLogger(InterfaceWithNoCanonicalModifiers.class);",
+            "  }",
+            "",
+            "  interface InterfaceWithWrongVariableName {",
+            "    // BUG: Diagnostic contains:",
+            "    Logger BAZ = LoggerFactory.getLogger(InterfaceWithWrongVariableName.class);",
+            "  }",
+            "",
+            "  interface WrongArgumentGetLoggerInterface {",
+            "    // BUG: Diagnostic contains:",
+            "    Logger LOG = LoggerFactory.getLogger(A.class);",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  void replacement() {
+    BugCheckerRefactoringTestHelper.newInstance(Slf4jLogDeclaration.class, getClass())
+        .addInputLines(
+            "A.java",
+            "import org.slf4j.Logger;",
+            "import org.slf4j.LoggerFactory;",
+            "",
+            "class A {",
+            "  static Logger foo = LoggerFactory.getLogger(A.class);",
+            "",
+            "  void m() {",
+            "    String unused = toString();",
+            "    foo.trace(\"foo\");",
+            "  }",
+            "",
+            "  static class NestedClass {",
+            "    void m() {",
+            "      foo.trace(\"foo\");",
+            "    }",
+            "  }",
+            "",
+            "  static class WrongArgumentGetLogger {",
+            "    private static final Logger LOG = LoggerFactory.getLogger(String.class);",
+            "  }",
+            "",
+            "  static class WrongArgumentOverloadedGetLogger {",
+            "    // BUG: Diagnostic contains:",
+            "    private static final Logger LOG = LoggerFactory.getLogger(\"OverloadedGetLogger\");",
```

Comment:
On a second look, I'm not really sure how we should handle these string overloads... There are quite some different ways of using them, if we look outside of [Picnic to usages](https://github.com/search?q=LoggerFactory.getLogger%28%22&type=code). Within Picnic we have a few like `Something.NameOfclass`, so not sure what we want to do there... 

CC: @Stephan202 What do you think?

Reply:
> ...if we look outside of [Picnic to usages](https://github.com/search?q=LoggerFactory.getLogger%28%22&type=code)....

You're missing the filter on [Java code](https://github.com/search?q=LoggerFactory.getLogger%28%22+language%3AJava&type=code&l=Java) for this

Reply:
Good question @rickie. It does seem safer to ignore these cases, or or put this logic under a flag. But I'm not sure there's much use in the latter, because users should favour the `Class` overload anyway when possible. So any string variant should likely be considered "custom". TL;DR: seems better to cover only `Class`-accepting `getLogger` calls. WDYT?

Reply:
> But I'm not sure there's much use in the latter, because users should favour the Class overload anyway when possible.

Can / should we flag this as part of the check?